### PR TITLE
Issue #23027:global uom

### PR DESCRIPTION
**Pull request by garyhgohoos**

### CODE REVIEW

File: guiclient/creditMemoItem.cpp [hunk 1]
```diff
@@ -748,8 +718,92 @@ void creditMemoItem::sTaxDetail()
...
+    
+    ParameterList params;
+    params.append("uomtype", "Selling");
+    params.append("item_id", _item->id());
+    
+    // Include Global UOMs
+    if (_privileges->check("MaintainUOMs"))
+    {
+      params.append("includeGlobal", true);
+      params.append("global", tr("-Global"));
```
Comment (gilmoskowitz) [inline guiclient/creditMemoItem.cpp]: typo

File: guiclient/creditMemoItem.cpp [hunk 2]
```diff
@@ -782,6 +836,42 @@ void creditMemoItem::sPriceUOMChanged()
...
+      _pricingUOM->setId(savepriceuomid);
+      connect(_qtyUOM,     SIGNAL(newID(int)), this, SLOT(sQtyUOMChanged()));
+      connect(_pricingUOM, SIGNAL(newID(int)), this, SLOT(sPriceUOMChanged()));
+    }
+    else
+    {
+      _pricingUOM->setId(_invuomid);
+    }
+  }
+
```
Comment (gilmoskowitz) [inline guiclient/creditMemoItem.cpp]: This duplicated code should be in a separate (private) method.

File: guiclient/invoiceItem.cpp [hunk 3]
```diff
@@ -635,6 +689,42 @@ void invoiceItem::sPriceUOMChanged()
...
+      _pricingUOM->setId(savepriceuomid);
+      connect(_qtyUOM,     SIGNAL(newID(int)), this, SLOT(sQtyUOMChanged()));
+      connect(_pricingUOM, SIGNAL(newID(int)), this, SLOT(sPriceUOMChanged()));
+    }
+    else
+    {
+      _pricingUOM->setId(_invuomid);
+    }
+  }
+
```
Comment (gilmoskowitz) [inline guiclient/invoiceItem.cpp]: Same here - private method to reduce code duplication.

File: guiclient/creditMemoItem.cpp (hunk 1 above)
Comment (garyhgohoos) [inline guiclient/creditMemoItem.cpp]: I don't see the typo???

Comment (gilmoskowitz) [inline guiclient/creditMemoItem.cpp]: Is the `-` in `-Global` here and elsewhere desired?

Comment (garyhgohoos) [inline guiclient/creditMemoItem.cpp]: I am appending the value of the "global" parameter to uom_name to differentiate global uoms from uoms already defined for an item.  for example, "KG-Global"

**Pull request merged**